python/tk_desktop/desktop_window.py

```diff
@@ -1127,6 +1124,37 @@ def __get_server_version(self, connection):
         sg_patch_ver = connection.server_info["version"][2]
         return sg_major_ver, sg_minor_ver, sg_patch_ver
 
+    def engine_startup_error(self, error, tb):
+        """
+        Handle an error starting up the engine for the app proxy.
+
+        :param error: Exception object that was raised during bootstrap.
+        :param tb: Traceback of the exception raised during bootstrap.
+        """
+        trigger_project_config = False
+        if isinstance(error, sgtk.TankEngineInitError):
+            # match directly on the error message until something less fragile can be put in place
+            if error.message.startswith("Cannot find an engine instance tk-desktop"):
```

Comment:
is this from core? Could we add a new exception type subclassing from TankEngineInitError - perhaps as part of https://github.com/shotgunsoftware/tk-core/pull/475 ?

If that is complicated, we should add a comment in core pointing out that changing that string may break desktop. But would be great if we could avoid it!